src/System.Data.Common/tests/System/Data/Common/DbDataReaderTest.cs

```diff
@@ -28,113 +28,470 @@ namespace System.Data.Tests.Common
-
-			 _dataReader.Read();
-			Stream stream = _dataReader.GetStream(testColOrdinal);
-			Assert.NotNull(stream);
-
-			//Read stream content to byte buffer
-			int data_length = stream.Read(buffer, 0, buffer.Length);
-
-			//Verify that content is expected
-			byte[] expected = new byte[] { 0xde, 0xad, 0xbe, 0xef };
-			Assert.Equal(expected.Length, data_length);
-			for (int i = 0; i < expected.Length; i++)
-			{
-				Assert.Equal(expected [i], buffer [i]);
-			}
-
-			//Get DBNull value stream
-			Assert.True(_dataReader.Read());
-			stream = _dataReader.GetStream(testColOrdinal);
-			Assert.Equal(0, stream.Length);
-
-			//Get single byte value stream
-			Assert.True(_dataReader.Read());
-			stream = _dataReader.GetStream(testColOrdinal);
-			expected = new byte[] { 0x00 };
-			Assert.Equal(expected.Length, stream.Length);
-			Assert.Equal(expected [0], stream.ReadByte());
-		}
-
-		[Fact]
-		public void GetTextReader()
-		{
-			int testColOrdinal = 0;
-
-			//Read first row
-			 _dataReader.Read();
-			TextReader textReader = _dataReader.GetTextReader(testColOrdinal);
-			Assert.NotNull(textReader);
-
-			string txt = textReader.ReadToEnd();
-			Assert.Equal("row_1", txt);
-
-			//Move to second row
-			Assert.True(_dataReader.Read());
-			textReader = _dataReader.GetTextReader(testColOrdinal);
-			txt = textReader.ReadToEnd();
-			Assert.Equal("row_2", txt);
-
-			//Move to third row
-			Assert.True(_dataReader.Read());
-			textReader = _dataReader.GetTextReader(testColOrdinal);
-			txt = textReader.ReadToEnd();
-			Assert.Equal("row_3", txt);
-
-			Assert.False(_dataReader.Read());
-		}
+            Assert.Equal(4, testData.Rows.Count);
+        }
+
+        [Fact]
+        public void GetBooleanByColumnNameTest()
+        {
+            SkipRows(3);
+
+            var expected = true;
+
+            // The row after rowsToSkip
+            _dataReader.Read();
+
+            var actual = _dataReader.GetBoolean("boolean_col");
+
+            Assert.Equal(expected, actual);
+        }
+
+        [Fact]
+        public void GetByteByColumnNameTest()
+        {
+            SkipRows(3);
+
+            var expected = (byte)0x00;
+
+            // The row after rowsToSkip
+            _dataReader.Read();
+
+            var actual = _dataReader.GetByte("byte_col");
+
+            Assert.Equal(expected, actual);
+        }
+
+        [Fact]
+        public void GetBytesByColumnNameTest()
+        {
+            SkipRows(3);
+
+            var expected = new byte[] { 0xAD, 0xBE };
+
+            // The row after rowsToSkip
+            _dataReader.Read();
+
+            var actual = new byte[1024];
+            var length = _dataReader.GetBytes("binary_col", 1, actual, 0, 2);
+
+            Assert.Equal(expected.Length, length);
+            for (int i = 0; i < expected.Length; i++)
+            {
+                Assert.Equal(expected[i], actual[i]);
+            }
+        }
+
+        [Fact]
+        public void GetCharByColumnNameTest()
```

Comment:
This is going to throw on SqlClient because it doesn't support [`GetChar`](https://github.com/dotnet/corefx/blob/f26fb9c2a6fe52a5cdb9309f1b7dbe14e1e667b0/src/System.Data.SqlClient/src/System/Data/SqlClient/SqlDataReader.cs#L1844). I don't know what behaviour odbc has.
I don't think the test is wrong but it may be worth investigating if this test is run as part of the SqlClient tests and if so whether that behaviour needs to be taken into account somehow.

Reply:
The test itself runs using a fake DB Reader (`DbDataReaderMock`). It should not impact or be impacted by `SqlDataReader`. Also, `SqlDataReader` will throw the same exception with `GetChar(string name)` that it throws with `GetChar(int i)`; which it the expected behaviour.

Unless I misunderstood your comment or misunderstood how things work :)

Reply:
Ah, the Common mislead me. There's some stuff in src\Common that gets included as files into other projects. My bad. Looks good.